components/advanced/babel-plugin.js

```diff
@@ -0,0 +1,165 @@
+import React from 'react';
+import SectionLayout from '../SectionLayout';
+import CodeBlock from '../CodeBlock';
+import Code from '../Code';
+import Note from '../Note';
+
+const installNPM = `
+npm install --save-dev babel-plugin-styled-components
+`.trim();
+
+const usage = `
+{
+  "plugins": ["styled-components"]
+}
+`.trim();
+
+const ssr = `
+{
+  "plugins": [
+    ["styled-components", {
+      "ssr": true
+    }]
+  ]
+}
+`.trim();
+
+const displayName = `
+{
+  "plugins": [
+    ["styled-components", {
+      "displayName": false
+    }]
+  ]
+}
+`.trim();
+
+const preprocess = `
+{
+  "plugins": [
+    ["styled-components", {
+      "preprocess": true
+    }]
+  ]
+}
+`.trim();
+
+const minify = `
+{
+  "plugins": [
+    ["styled-components", {
+      "minify": false
+    }]
+  ]
+}
+`.trim();
+
+const BabelPlugin = () =>
+  <SectionLayout title="Babel Plugin" labels={['v2']}>
+    <p>
+      This plugin adds support for server-side rendering, for minification of
+      styles and gives you a nicer debugging experience when using{' '}
+      <Code>styled-components</Code>.
```

Comment:
You can leave out "when using styled-components" as that's quite obvious. But do add a paragraph that it's only for the web, not for React Native please :wink: